ethsigner/app/src/main/java/tech/pegasys/ethsigner/EthSignerApp.java

```diff
@@ -35,8 +38,26 @@ public static void main(final String... args) {
     LOG.debug("Configuration = {}", config);
     LOG.info("Version = {}, ", ApplicationInfo.version());
 
+    // create a signer based on the configuration provided
+    TransactionSigner signer = createTransactionSigner(config);
+
+    if (signer == null) {
+      LOG.error("Cannot create a signer from the given config: " + config.toString());
+      System.exit(-1);
+    }
+
     final tech.pegasys.ethsigner.core.EthSigner ethSigner =
-        new tech.pegasys.ethsigner.core.EthSigner(config, new RunnerBuilder());
+        new tech.pegasys.ethsigner.core.EthSigner(config, signer, new RunnerBuilder());
     ethSigner.run();
   }
+
+  private static TransactionSigner createTransactionSigner(CommandLineConfig config) {
```

Comment:
nit; input variable can have the `final` modifier

Reply:
done